rs/boundary_node/anonymization/backend/src/lib.rs

```diff
@@ -103,16 +159,39 @@ fn post_upgrade() {
 }
 
 #[ic_cdk::update]
-fn register(_pubkey: Vec<u8>) -> RegisterResponse {
-    unimplemented!()
+fn register(pubkey: Vec<u8>) -> RegisterResponse {
+    match REGISTRATOR.with(|v| v.borrow().register(&pubkey)) {
```

Comment:
Consider moving these matching statements into the `register()` function. This would make the canister method thinner and easier. Plus the unit test will cover all the logic, including error handling. Also consider implementing conversion of errors into each other.